account_bank_statement_import_online/models/online_bank_statement_provider.py

```diff
@@ -225,7 +238,9 @@ def _pull(self, date_since, date_until):
                     )
                 filtered_lines = []
                 for line_values in lines_data:
-                    date = fields.Datetime.from_string(line_values['date'])
+                    date = line_values['date']
+                    if isinstance(date, str):
+                        date = fields.Datetime.from_string(date)
```

Comment:
if `line_values['date']` is a `date`, you should convert in `datetime` if not after you compare `date` and `datetime`.
`from_string` make this

Reply:
Great point, thanks